content/build/context/dockerignore.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Dockerignore files
+description: A .dockerignore file lets you exclude files from the context using a glob pattern
+keywords: build, context, dockerignore, exclude, glob
+---
+
+You can use a `.dockerignore` file to exclude files or directories from the
+build context.
+
+```gitignore
+# .dockerignore
+node_modules
+bar
+```
+
+This helps avoid sending unwanted files and directories to the builder,
+improving build speed, especially when using a remote builder.
+
+## Filename and location
+
+When you run a build command, the build client looks for a file named
+`.dockerignore` in the root directory of the context. If this file exists, the
+files and directories that match patterns in the files are removed from the
+build context before it's sent to the builder.
+
+If you use multiple Dockerfiles, you can use different ignore-files for each
+Dockerfile. You do so using a special naming convention for the ignore-files.
+Place your ignore-file in the same directory as the Dockerfile, and prefix the
+ignore-file with the name of the Dockerfile, as shown in the following example.
+
+```text
+.
+├── index.ts
+├── src/
+├── docker
+│   ├── build.Dockerfile
+│   ├── build.Dockerfile.dockerignore
+│   ├── lint.Dockerfile
+│   ├── lint.Dockerfile.dockerignore
+│   ├── test.Dockerfile
+│   └── test.Dockerfile.dockerignore
+├── package.json
+└── package-lock.json
+```
+
+A Dockerfile-specific ignore-file takes precedence over the `.dockerignore`
+file at the root of the build context if both exist.
+
+## Syntax
+
+The CLI interprets the `.dockerignore` file as a newline-separated list of
```

Comment:
It was the CLI with the legacy builder iirc (cc @thaJeztah @tonistiigi) but since BuildKit it's the frontend. We could then cross link to https://docs.docker.com/build/dockerfile/frontend/#dockerfile-frontend?

Reply:
Maybe we should be less specific "who" interprets it

Reply:
Huh, I didn't know this either! So Buildx transacts the .dockerignore file and Dockerfile first, and then BuildKit responds with what context files it needs?

Reply:
and yes @thaJeztah :)